package/libpiksi/libpiksi/src/networking.c

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright (C) 2018 Swift Navigation Inc.
+ * Contact: Swift Navigation <dev@swiftnav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#include <arpa/inet.h>
+#include <sys/socket.h>
+#include <ifaddrs.h>
+#include <linux/if_link.h>
+
+#include <libpiksi/util.h>
+#include <libpiksi/networking.h>
+
+typedef s32 (*interface_map_fn_t)(struct ifaddrs *ifa, void *userdata);
+
+// refactored from original implementation in piksi_system_daemon by @axlan
+void map_network_interfaces(interface_map_fn_t map_fn, void *userdata)
```

Comment:
Looks like this can be static?

Reply:
Yes good catch. I was considering the mapping portion for a public interface for use elsewhere if needed but it seems like an unnecessary abstraction at this point since you still have to interact with the ifaddr references in the mapped function.